manticore/native/cpu/abstractcpu.py

```diff
@@ -985,6 +998,7 @@ def execute(self):
         """
         curpc = self.PC
         if self._delayed_event:
+            self._last_executed_pc = self._last_pc
```

Comment:
I think lines 1001-1008 could be safely replaced with a call to `self._publish_instruction_as_executed`